src/gui/editors/PianoRoll.cpp

```diff
@@ -4803,6 +4834,116 @@ void PianoRollWindow::ghostPatternSet( bool state )
 
 
 
+void PianoRollWindow::exportPattern()
+{
+	QString extFilter("XML Pattern (*.xpt *.xptz)");
+	FileDialog exportDialog(this, tr("Export pattern"), "", extFilter);
+
+	exportDialog.setAcceptMode(FileDialog::AcceptSave);
+
+	if (exportDialog.exec() == QDialog::Accepted &&
+		!exportDialog.selectedFiles().isEmpty() &&
+		!exportDialog.selectedFiles().first().isEmpty())
+	{
+		QString suffix = ConfigManager::inst()->value( "app",
+								"nommpz" ).toInt() == 0
+							? "xptz"
+							: "xpt" ;
+		exportDialog.setDefaultSuffix(suffix);
+
+		QString fullPath = exportDialog.selectedFiles()[0];
+		QString chosenSuffix = fullPath.section( '.', -1 );
+
+		bool compress = (chosenSuffix == "xpt") ? false : true;
+
+		if (savePatternXML(fullPath, compress) != 0)
+		{
+			TextFloat::displayMessage(tr("Export pattern failed"),
+				tr("No permission to write to %1").arg(fullPath),
+				embed::getIconPixmap("error"), 4000);
+			return;
+		}
+		TextFloat::displayMessage(tr("Export pattern success"),
+			tr("Pattern saved to %1").arg(fullPath),
+			embed::getIconPixmap("project_export"), 4000);
+	}
+}
+
+
+
+
+void PianoRollWindow::importPattern()
+{
+	// Overwrite confirmation.
+	if (!m_editor->m_pattern->empty() && QMessageBox::warning(NULL,
+				tr("Import pattern."),
+				tr("You are about to import a pattern, this will "
+					"overwrite your current pattern. Do you want to "
+					"continue?"),
+				QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes)
+				!= QMessageBox::Yes)
+	{
+		return;
+	}
+
+	FileDialog importDialog(this, tr("Open Pattern"), "",
+			tr("XML pattern file (*.xpt *.xptz)"));
+	importDialog.setFileMode(FileDialog::ExistingFiles);
+
+	if (importDialog.exec () == QDialog::Accepted &&
+		!importDialog.selectedFiles().isEmpty())
```

Comment:
```suggestion
	if (importDialog.exec() == QDialog::Accepted &&
		!importDialog.selectedFiles().isEmpty())
```

Extra space before parenthesis.

I also got curious, why does the export requires a `!selectedFiles().first().isEmpty()` check and the import doesn't? Is it because the file mode is `ExistingFiles`, so it will only accept the dialog with an existing file name?